### PR TITLE
chore(deps): React 18 migration

### PR DESCRIPTION
**Pull request by leSamo**
https://issues.redhat.com/browse/RHINENG-7797

I had to disable 3 flaky tests, but I made sure to manually check if the test cases still work. I will create a follow-up ticket to rewrite the tests.

### CODE REVIEW

**Comment by InsightsDroid** [comment]
## :exclamation: Outdated dependencies error
* Installed version of @redhat-cloud-services/frontend-components-config (4.6.40) is older than required version (6.0.0)

**Comment by InsightsDroid** [comment]
## :exclamation: Outdated dependencies error
* Installed version of @redhat-cloud-services/frontend-components-config (4.6.40) is older than required version (6.0.0)

File: src/bootstrap-dev.tsx [hunk 1]
```diff
@@ -1,10 +1,11 @@
 import React from 'react';
```
Comment (bastilian) [inline src/bootstrap-dev.tsx:1]: These `bootstrap` files shouldn't be needed anymore if the `fec` cli and shared webpack config are use. Only the modules exposed in the `fec.config.js` should be needed. 

The `entry(-dev).js` files are also not really needed anymore, but the `fec` cli is looking for them still, so we can't remove those, but we can point the imports they include just to `AppEntry.js`.

File: src/pages/PolicyDetail/__tests__/PolicyDetail.test.tsx [hunk 2]
```diff
@@ -345,9 +345,9 @@ describe('src/Pages/PolicyDetail/PolicyDetail', () => {
         expect(screen.getByDisplayValue(/Copy of Not arch x86_64/i)).toBeVisible();
     });
 
-    it('Duplicates navigates to the new policy url', async () => {
+    it.skip('Duplicates navigates to the new policy url', async () => {
```
Comment (bastilian) [inline src/pages/PolicyDetail/__tests__/PolicyDetail.test.tsx:348]: At least these two `skip`ed tests seem to be fine when running the tests locally. Only the provider test seems to fail. Are they flaky or can we not skip the?

**Review by bastilian — APPROVED** [review]
LGTM! Only the two minor things with the tests and bootstrap files, but that is not strictly necessary , at least now. Thank you @leSamo!

File: src/pages/PolicyDetail/__tests__/PolicyDetail.test.tsx (hunk 2 above)
Comment (leSamo) [reply src/pages/PolicyDetail/__tests__/PolicyDetail.test.tsx:348]: I tried again, but they are flaky on my device

`npm run test -- policydetail.test`

1st run
> Test Suites: 1 failed, 1 total
> Tests:       1 failed, 26 passed, 27 total

2nd run
> Test Suites: 1 failed, 1 total
> Tests:       2 failed, 25 passed, 27 total

3rd run
> Test Suites: 1 failed, 1 total
> Tests:       1 failed, 26 passed, 27 total

4th run
> Test Suites: 1 failed, 1 total
> Tests:       1 failed, 26 passed, 27 total

5th run:
> Test Suites: 1 failed, 1 total
> Tests:       2 failed, 25 passed, 27 total

**Comment by InsightsDroid** [comment]
## :exclamation: Outdated dependencies error
* Installed version of @redhat-cloud-services/frontend-components-config (4.6.40) is older than required version (6.0.0)

**Pull request merged**